flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/WriteAheadSinkTestBase.java

```diff
@@ -158,17 +160,151 @@ public void testDataDiscardingUponRestore() throws Exception {
 		testHarness = new OneInputStreamOperatorTestHarness<>(sink);
 
 		testHarness.setup();
-		testHarness.restore(latestSnapshot);
+		testHarness.initializeState(latestSnapshot);
 		testHarness.open();
 
 		for (int x = 0; x < 20; x++) {
 			testHarness.processElement(new StreamRecord<>(generateValue(elementCounter, 2)));
 			elementCounter++;
 		}
 
-		testHarness.snapshotLegacy(snapshotCount++, 0);
+		testHarness.snapshot(snapshotCount++, 0);
 		testHarness.notifyOfCompletedCheckpoint(snapshotCount - 1);
 
-		verifyResultsDataDiscardingUponRestore(testHarness, sink);
+		verifyResultsDataDiscardingUponRestore(sink);
+	}
+
+	@Test
+	public void testScalingDown() throws Exception {
+		S sink1 = createSink();
+		OneInputStreamOperatorTestHarness<IN, IN> testHarness1 =
+			new OneInputStreamOperatorTestHarness<>(sink1, 10, 2, 0);
+		testHarness1.open();
+
+		S sink2 = createSink();
+		OneInputStreamOperatorTestHarness<IN, IN> testHarness2 =
+			new OneInputStreamOperatorTestHarness<>(sink2, 10, 2, 1);
+		testHarness2.open();
+
+		int elementCounter = 1;
```

Comment:
we should let this start at 0; i just ended up double-checking that the count add up since the very loop goes to 34 even though we only write 33 elements.